helix-view/src/editor.rs

```diff
@@ -71,6 +71,70 @@ where
     )
 }
 
+// Accept either a simple array of strings as gutter defaults or
+// a [editor.gutters] block
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum GutterConfigOption {
+    Simple(Vec<GutterType>),
+    Detailed(GutterConfig),
+}
```

Comment:
Rather than juggling two types, you could use a serde visitor to parse into just the `GutterConfig` type: https://github.com/helix-editor/helix/pull/3798#issuecomment-1242947382

Also see where we use this pattern in `keymap.rs` https://github.com/helix-editor/helix/blob/fa436fa680b5bcf6d46666a03ae23b94a74e0414/helix-term/src/keymap.rs#L154-L209 (although it's used there for better error messages https://github.com/helix-editor/helix/pull/3931)

Reply:
I had a [`Deserialize` implementation](https://github.com/helix-editor/helix/blob/ea094ed143d6d23d34d1d667650fcf57025c48b6/helix-view/src/editor.rs#L92-L145) previously (not as elegant as using a Visitor, but functionally similar), but wasn't thrilled with how it came out. I'm still grappling with the right way to do this. 

With both the custom `Deserialize` and `Visitor` patterns, I can't find a way to re-use the derived `Deserialize` - which is still what I want to use for deserializing maps, with the added custom `visit_seq`. Is it possible to still derive `visit_map` (or equivalent) while implementing a custom `visit_seq`? It _feels_ like this should be possible, but I'm not digging anything up while reading through the `serde` docs.

I'll continue to hunt around for an elegant solution here, but any nudges in the right direction would be appreciated.

Reply:
The `deserialize_with` option in serdw is intended for these kinds of situations

Reply:
Cheers - thanks @pascalkuthe. I'll look into that.

Reply:
Thanks both for the pointers. I'm much happier with where it's at now.